features/270-next-skycli/api.md

```diff
@@ -0,0 +1,285 @@
+## Feature Overview
+
+In next, we will have cli tools for user to manage the cluster's applications.
+Since we will only have skycli in the first version, skycli should be able to
+do almost all the operations of v1 portal.
+
+The following spec will cover the EE version first.
+
+
+## User scenarios
+
+- Cluster admin can create cluster user
+- Cluster admin assign user to app
+- Cluster user can create app
+- Cluster user can view, update and delete their apps config
+- Cluster user deploy CF to their apps
+- Cluster user can update apps custom domain setting, add new custom domain to app
+- Cluster admin can do everything in the cluster
+
+## Commands
+
+- skycli login (alisa of `skycli auth login`)
+
+- skycli logout (alisa of `skycli auth logout`)
+
+- skycli auth
+    - [skycli auth login](#skycli-auth-login)
+    - [skycli auth logout](#skycli-auth-logout)
+
+- skycli config
```

Comment:
I'm aware of these set of config command looks similar (and work similarly) to kubectl, which could be an advantage, however, as Skygear would works kind of "repo" based, would it be better if we make skycli adopt config (which app, which cluster), depends on the working directory's `.skyclirc` file? It also has the advantage of developers can check-in their .skyclirc file, and as long as the user is logged in, running `skycli deploy` can always deploy to the correct repo.

Reply:
I added a section to tell where is the information store.

https://github.com/SkygearIO/features/blob/24ed5a631ebda8e8ed760c58525c1c79a55d5de5/features/270-next-skycli/api.md#config-files-structure

I prefer to store account and cluster info in user directory. And store app in repo directory, `set-app` and `set-cluster-server` will update different files.